addons/html_editor/static/src/main/table/table_plugin.js

```diff
@@ -640,6 +640,10 @@ export class TablePlugin extends Plugin {
             ) {
                 // Handle selecting an empty cell.
                 this.selectTableCells(selection);
+                this.dependencies.toolbar.updateToolbar(
+                    this.dependencies.selection.getSelectionData(),
+                    { forceOpen: true }
+                );
```

Comment:
I don't think you need `updateToolbar` here there is already an `mouseup` event listener in toolbar plugin which does `updateToolbar` and in `updateToolbarVisibility` `shouldBeVisible` should be true for you case, meaning when seleciton is collapsed and the td has class `o_selected_td`.